src/gilded_rose.js

```diff
@@ -15,46 +15,66 @@ items.push(new Item('Conjured Mana Cake', 3, 6));
 
 function update_quality() {
   for (var i = 0; i < items.length; i++) {
-    if (items[i].name != 'Aged Brie' && items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-      if (items[i].quality > 0) {
-        if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-          items[i].quality = items[i].quality - 1
-        }
+
+    if (items[i].name === '+5 Dexterity Vest' || items[i].name === 'Elixir of the Mongoose') {
```

Comment:
The problem with the original design is that multiple `if`'s and nested `if`'s as a design approach it not extensible. It is too hard to understand. 

Your refactoring left the original design approach in place, meaning that future changes will still be hard and the developer will have to understand the implementation entirely before being able to make changes. 

Refactoring to a better design will make it easier to add new functionality. That's where the `make the change easy, then make the easy change` concept comes from. Refactor to an extensible design (`make the change easy`) then  implement the new feature now that it's easy to add (`make the easy change`)